### DISCUSSION TITLE
If the solution is affected by the running time

### DISCUSSION THREAD

**Discussion by zzzZ-zy** [coin-or/Ipopt#672] [Q&A] [answered]
When I was solving a problem, I found that if I change the print_Level, whether to do derivative_test, or whether to redirect the results to a specific file can affect the solution results. Sometimes it can even affect whether the solution can be successfully solved. I guess these types of operations can affect the runtime, which may be the reason for the impact on the solution results.
I have reviewed the relevant discussions [Randomness in IPOPT solve #469](https://github.com/coin-or/Ipopt/discussions/469), but I have all been running on the same machine.

May I ask if this is related to my code writing or is there a problem with IPOPT itself? If there are some mistakes in my code, which part may causes this situation? How to solve it?
This is Ipopt version 3.14.12, running with linear solver ma27.
Thank you very much!

**Comment by svigerske** [comment, marked as answer]
If Ipopt stops on a timelimit, then you may of course get different end results when repeating the runs. But everything before the timelimit is reached should be unaffected.

Enabling the derivate tests means additional calls to your function/Jacobian/Hessian evaluation routines. You need to make sure that calls to these routines do not affect subsequent calls.

Changes to a print_level or so should not have any effect. Getting different results in this case is sometimes a symptom of using uninitialized memory somewhere. You may want to run under a tool like [valgrind](https://valgrind.org/) to check this.
